sources/modal/components/CustomModal.tsx

```
@@ -9,24 +9,43 @@ interface CustomModalProps {
 
+type CommandPaletteExternalProps = Omit<React.ComponentProps<typeof CommandPalette>, 'onClose'>;
+
 export function CustomModal({ config, onClose }: CustomModalProps) {
     const Component = config.component;
     
     // Use special modal wrapper for CommandPalette with animation support
     if (Component === CommandPalette) {
         return <CommandPaletteWithAnimation config={config} onClose={onClose} />;
     }
+
+    const handleClose = React.useCallback(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)